### PR TITLE
Fix extra new line in ATT&CK-coverage.md

### PR DESCRIPTION
**Pull request by shashank-elastic**
# Pull Request

*Issue link(s)*: PR(s) are being raised for newline with no actual changes 
Example - https://github.com/elastic/detection-rules/pull/4211

## Summary - What I changed

- Remove the additional new line, so PR is created only when content has changed
- Remove spaces from the header content so we wont have issues like in PR https://github.com/elastic/detection-rules/pull/4264/

## How To Test

- Unit Test should pass 
- We have Ran a workflow from the private branch https://github.com/elastic/detection-rules/actions/runs/11742727980/job/32714051302?pr=4263
- Only Updates are seen as in this PR with no header content changes https://github.com/elastic/detection-rules/pull/4265/files

## Checklist

- [x] Added a label for the type of pr: `bug`, `enhancement`, `schema`, `maintenance`, `Rule: New`, `Rule: Deprecation`, `Rule: Tuning`, `Hunt: New`, or `Hunt: Tuning` so guidelines can be generated
- [x] Added the `meta:rapid-merge` label if planning to merge within 24 hours
- [ ] Secret and sensitive material has been managed correctly
- [ ] Automated testing was updated or added to match the most common scenarios
- [ ] Documentation and comments were added for features that require explanation

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by protectionsmachine** [comment]
## Bug - Guidelines

These guidelines serve as a reminder set of considerations when addressing a bug in the code.

### Documentation and Context

- [ ] Provide detailed documentation (description, screenshots, reproducing the bug, etc.) of the bug if not already documented in an issue.
- [ ] Include additional context or details about the problem.
- [ ] Ensure the fix includes necessary updates to the release documentation and versioning.

### Code Standards and Practices

- [ ] Code follows established design patterns within the repo and avoids duplication.
- [ ] Code changes do not introduce new warnings or errors.
- [ ] Variables and functions are well-named and descriptive.
- [ ] Any unnecessary / commented-out code is removed.
- [ ] Ensure that the code is modular and reusable where applicable.
- [ ] Check for proper exception handling and messaging.

### Testing

- [ ] New unit tests have been added to cover the bug fix or edge cases.
- [ ] Existing unit tests have been updated to reflect the changes.
- [ ] Provide evidence of testing and detecting the bug fix (e.g., test logs, screenshots).
- [ ] Validate that any rules affected by the bug are correctly updated.
- [ ] Ensure that performance is not negatively impacted by the changes.
- [ ] Verify that any release artifacts are properly generated and tested.

### Additional Checks

- [ ] Ensure that the bug fix does not break existing functionality.
- [ ] Review the bug fix with a peer or team member for additional insights.
- [ ] Verify that the bug fix works across all relevant environments (e.g., different OS versions).
- [ ] Confirm that all dependencies are up-to-date and compatible with the changes.
- [ ] Confirm that the proper version label is applied to the PR `patch`, `minor`, `major`.

File: detection_rules/devtools.py [hunk 1]
```diff
@@ -836,20 +836,19 @@ def raw_permalink(raw_link):
...
+generated below. You can also include multiple across tabs in a single session, though it is not
+advisable to upload _all_ of them as it will likely overload your browsers resources.
 
-        ## Current rule coverage
+## Current rule coverage
 
-        The source files for these links are regenerated with every successful merge to main. These represent
-        coverage from the state of rules in the `main` branch.
+The source files for these links are regenerated with every successful merge to main. These represent
+coverage from the state of rules in the `main` branch.
```
Comment (eric-forte-elastic) [inline detection_rules/devtools.py:851]: Nit, technically the only change that is necessary to address the bug is the change on line(s) 839 and 840 -> 839 right?

Any particular reasoning to change this formatting here?

Comment (shashank-elastic) [reply detection_rules/devtools.py:851]: With the removal of new line it added extra spaces like in this PR https://github.com/elastic/detection-rules/pull/4211 pinned in the testing section so we just reformatted so header section remains the same!

Comment (shashank-elastic) [reply detection_rules/devtools.py:851]: image

Comment (eric-forte-elastic) [reply detection_rules/devtools.py:851]: I follow the logic, and think that makes since. Just for reference, it looks like that PR does not have a an extra spacing issue, just the initial newline issue? Perhaps its the wrong link or I am missing something?

image

Comment (shashank-elastic) [reply detection_rules/devtools.py:851]: Yes, but when we removed the extra line, it caused more spacing issues like seen above and had to remove those spaces as well.

Comment (shashank-elastic) [reply detection_rules/devtools.py:851]: Updated the summary with the right PR which caused spacing issues - https://github.com/elastic/detection-rules/pull/4264/files

Comment (eric-forte-elastic) [reply detection_rules/devtools.py:851]: Appears like this might be the better link for that https://github.com/elastic/detection-rules/pull/4264/files :slightly_smiling_face:

**Review by terrancedejesus — APPROVED** [review]
